php/docker/templates/nginx-vhost-conf.d/80-index.conf

```
@@ -1,12 +1,73 @@
 # vim: set ft=nginx:
 
+{{- $truthStrings := list "true" "on" "yes" }}
+
 location / {
     try_files $uri $uri/ /index.php$is_args$args;
 }
 
 location ~ \.php$ {
+    {{ if (has (default "off" .Env.PAGE_CACHE | lower) $truthStrings) }}
+    set $skip_cache 0;
+
+    # POST requests and urls with a query string should always go to PHP
+    if ($request_method = POST) {
+        set $skip_cache 1;
+    }
+    if ($query_string != "") {
+        set $skip_cache 1;
+    }
+
+    # Don't cache uris containing the following segments
+    if ($request_uri ~* "/wp-admin/|/xmlrpc.php|wp-.*.php|/feed/|index.php|sitemap(_index)?.xml") {
+        set $skip_cache 1;
+    }
+
+    # Don't use the cache for logged in users or recent commenters
+    if ($http_cookie ~* "comment_author|wordpress_[a-f0-9]+|wp-postpass|wordpress_no_cache|wordpress_logged_in") {
+        set $skip_cache 1;
+    }
+
+    set $key "nginx-cache:https$request_method$host$request_uri";
+    try_files $uri =404;
+
+    srcache_fetch_skip $skip_cache;
+    srcache_store_skip $skip_cache;
+
+    # https://github.com/openresty/srcache-nginx-module#srcache_response_cache_control
+    srcache_response_cache_control off;
+
+    set_escape_uri $escaped_key $key;
+
+    srcache_fetch GET /redis-fetch $key;
+    srcache_store PUT /redis-store key=$escaped_key&exptime=120;
```

@calind  suggested that we should do it more pluggable and to let the user if possible define it's backend cache configuration if needed.
```suggestion
    srcache_fetch GET /.stack-cache-fetch $key;
    srcache_store PUT /.stack-cache-store key=$escaped_key&exptime=120;
```